astropy/coordinates/representation/geodetic.py

```diff
@@ -10,6 +11,37 @@
 from .cartesian import CartesianRepresentation
 
 
+class WestLongitudeMixin:
+    def to_cartesian(self):
+        self.lon = self.lon.value
+        return super().to_cartesian()
+
+    @classmethod
+    def from_cartesian(cls, cart):
+        angular = super().from_cartesian(cart)
+        return cls(
```

Comment:
Here, I'd again suggest to replace in-place rather than create a new instance, i.e.,
```
# Change sign of longitude in-place, possibly re-wrapping.
lon = angular._lon
np.negative(lon, out=lon)
if lon.wrap_angle != 180*u.deg:
    lon._wrap_at(lon.wrap_angle)
return angular
```